tests/acceptance/features/bootstrap/RoleAssignmentContext.php

```diff
@@ -0,0 +1,245 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * ownCloud
+ *
+ * @author Viktor Scharf <v.scharf@owncloud.com>
+ * @copyright Copyright (c) 2022 Viktor Scharf v.scharf@owncloud.com
+ */
+
+use Behat\Behat\Context\Context;
+use GuzzleHttp\Exception\GuzzleException;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use PHPUnit\Framework\Assert;
+
+require_once 'bootstrap.php';
+
+/**
+ * Context for the TUS-specific steps using the Graph API
+ */
+class RoleAssignmentContext implements Context {
+
+	/**
+	 * @var FeatureContext
+	 */
+	private FeatureContext $featureContext;
+
+	/**
+	 * @var SpacesContext
+	 */
+	private SpacesContext $spacesContext;
+
+	/**
+	 * @var string
+	 */
+	private string $baseUrl;
+
+	/**
+	 * @var string
+	 */
+	private string $setttingsUrl = '/api/v0/settings/';
+
+	/**
+	 * This will run before EVERY scenario.
+	 * It will set the properties for this object.
+	 *
+	 * @BeforeScenario
+	 *
+	 * @param BeforeScenarioScope $scope
+	 *
+	 * @return void
+	 */
+	public function before(BeforeScenarioScope $scope): void {
+		// Get the environment
+		$environment = $scope->getEnvironment();
+		// Get all the contexts you need in this context from here
+		$this->featureContext = $environment->getContext('FeatureContext');
+		$this->spacesContext = $environment->getContext('SpacesContext');
+		$this->tusContext = $environment->getContext('RoleAssignmentContext');
+		$this->baseUrl = \trim($this->featureContext->getBaseUrl(), "/");
+	}
+
+	/**
+	 * @When /^user "([^"]*)" tries to get all existing roles$/
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function sendRequestGettingRoles(string $user): void {
+		$fullUrl = $this->baseUrl . $this->setttingsUrl . "roles-list";
+		$this->featureContext->setResponse($this->spacesContext->sendPostRequestToUrl($fullUrl, $user, $this->featureContext->getPasswordForUser($user), "{}"));
+	}
+
+	/**
+	 * @param string $user
+	 * @param string $userId
+	 * @param string $roleId
+	 *
+	 * @return void
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function sendRequestAssignmentsAdd(string $user, string $userId, string $roleId): void {
```

Comment:
Always good to reflect the function behavior in its name. (Descriptive function name)
```suggestion
	public function <what-function-does>(string $user, string $userId, string $roleId): void {
```